libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/plugin/SalesforceNetworkPlugin.java

```diff
@@ -125,7 +125,14 @@ public void onSuccess(RestRequest request, RestResponse response) {
                     try {
                         // Not a 2xx status
                         if (!response.isSuccess()) {
-                            callbackContext.error(response.asString());
+                            JSONObject responseObject = new JSONObject();
+                            responseObject.put("allHeaders", new JSONObject(response.getAllHeaders()));
+                            responseObject.put("statusCode", response.getStatusCode());
+                            responseObject.put("body", response.asString());
```

Comment:
Shouldn't this value should be `response.asJsonObject()`?

Reply:
Body might not be json - so giving the raw body (string) is probably safer.

Reply:
Okay, fair enough. We can leave this as-is.

Reply:
I forgot that on iOS we check to see if it's a JSON object or array and then default to a string if not, I'll update this to match